openbsd symlink mode test failure

```

self = <borg.testsuite.archiver.ArchiverTestCase testMethod=test_basic_functionality>

    def test_basic_functionality(self):
        have_root = self.create_test_files()
        # fork required to test show-rc output
        output = self.cmd('init', '--encryption=repokey', '--show-version', '--show-rc', self.repository_location, fork=True)
        self.assert_in('borgbackup version', output)
        self.assert_in('terminating with success status, rc 0', output)
        self.cmd('create', self.repository_location + '::test', 'input')
        output = self.cmd('create', '--stats', self.repository_location + '::test.2', 'input')
        self.assert_in('Archive name: test.2', output)
        self.assert_in('This archive: ', output)
        with changedir('output'):
            self.cmd('extract', self.repository_location + '::test')
        list_output = self.cmd('list', '--short', self.repository_location)
        self.assert_in('test', list_output)
        self.assert_in('test.2', list_output)
        expected = [
            'input',
            'input/bdev',
            'input/cdev',
            'input/dir2',
            'input/dir2/file2',
            'input/empty',
            'input/file1',
            'input/flagfile',
        ]
        if are_fifos_supported():
            expected.append('input/fifo1')
        if are_symlinks_supported():
            expected.append('input/link1')
        if are_hardlinks_supported():
            expected.append('input/hardlink')
        if not have_root:
            # we could not create these device files without (fake)root
            expected.remove('input/bdev')
            expected.remove('input/cdev')
        if has_lchflags:
            # remove the file we did not backup, so input and output become equal
            expected.remove('input/flagfile')  # this file is UF_NODUMP
            os.remove(os.path.join('input', 'flagfile'))
        list_output = self.cmd('list', '--short', self.repository_location + '::test')
        for name in expected:
            self.assert_in(name, list_output)
>       self.assert_dirs_equal('input', 'output/input')

src/borg/testsuite/archiver.py:385: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
src/borg/testsuite/__init__.py:151: in assert_dirs_equal
    self._assert_dirs_equal_cmp(diff, **kwargs)
src/borg/testsuite/__init__.py:192: in _assert_dirs_equal_cmp
    self.assert_equal(d1, d2)
E   AssertionError: Lists differ: ['link1', 41453, 1000, 0, None, 1, 0, 1484429950428914974, {}] != ['link1', 41408, 1000, 0, None, 1, 0, 1484429950428914974, {}]
E   
E   First differing element 1:
E   41453
E   41408
E   
E   - ['link1', 41453, 1000, 0, None, 1, 0, 1484429950428914974, {}]
E   ?              ^^
E   
E   + ['link1', 41408, 1000, 0, None, 1, 0, 1484429950428914974, {}]
E   ?              ^^
```

same thing for the openbsd 6.4 box.

strange: the automated tox run on vagrant box startup doesn't show a failure, but when I run the test manually, it shows up:
```
(borg-env) -bash-4.4$ pytest -vk test_basic
=============================================================================================== test session starts ===============================================================================================
platform openbsd6 -- Python 3.6.6, pytest-4.4.2, py-1.8.0, pluggy-0.11.0 -- /vagrant/borg/borg-env/bin/python3
cachedir: .pytest_cache
benchmark: 3.2.2 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
Tests enabled: root, symlinks, hardlinks, atime/mtime, modes
Tests disabled: BSD flags, fuse
rootdir: /vagrant/borg/borg, inifile: setup.cfg
plugins: xdist-1.28.0, forked-1.0.2, cov-2.7.1, benchmark-3.2.2
collected 1353 items / 1348 deselected / 5 selected                                                                                                                                                               

src/borg/testsuite/archiver.py::ArchiverTestCase::test_basic_functionality FAILED                                                                                                                           [ 20%]
src/borg/testsuite/archiver.py::ArchiverTestCaseBinary::test_basic_functionality SKIPPED                                                                                                                    [ 40%]
src/borg/testsuite/archiver.py::RemoteArchiverTestCase::test_basic_functionality PASSED                                                                                                                     [ 60%]
src/borg/testsuite/archiver.py::DiffArchiverTestCase::test_basic_functionality PASSED                                                                                                                       [ 80%]
src/borg/testsuite/archiver.py::TestBuildFilter::test_basic PASSED                                                                                                                                          [100%]

==================================================================================================== FAILURES =====================================================================================================
____________________________________________________________________________________ ArchiverTestCase.test_basic_functionality ____________________________________________________________________________________

self = <borg.testsuite.archiver.ArchiverTestCase testMethod=test_basic_functionality>

    def test_basic_functionality(self):
        have_root = self.create_test_files()
        # fork required to test show-rc output
        output = self.cmd('init', '--encryption=repokey', '--show-version', '--show-rc', self.repository_location, fork=True)
        self.assert_in('borgbackup version', output)
        self.assert_in('terminating with success status, rc 0', output)
        self.cmd('create', '--exclude-nodump', self.repository_location + '::test', 'input')
        output = self.cmd('create', '--exclude-nodump', '--stats', self.repository_location + '::test.2', 'input')
        self.assert_in('Archive name: test.2', output)
        self.assert_in('This archive: ', output)
        with changedir('output'):
            self.cmd('extract', self.repository_location + '::test')
        list_output = self.cmd('list', '--short', self.repository_location)
        self.assert_in('test', list_output)
        self.assert_in('test.2', list_output)
        expected = [
            'input',
            'input/bdev',
            'input/cdev',
            'input/dir2',
            'input/dir2/file2',
            'input/empty',
            'input/file1',
            'input/flagfile',
        ]
        if are_fifos_supported():
            expected.append('input/fifo1')
        if are_symlinks_supported():
            expected.append('input/link1')
        if are_hardlinks_supported():
            expected.append('input/hardlink')
        if not have_root:
            # we could not create these device files without (fake)root
            expected.remove('input/bdev')
            expected.remove('input/cdev')
        if has_lchflags:
            # remove the file we did not backup, so input and output become equal
            expected.remove('input/flagfile')  # this file is UF_NODUMP
            os.remove(os.path.join('input', 'flagfile'))
        list_output = self.cmd('list', '--short', self.repository_location + '::test')
        for name in expected:
            self.assert_in(name, list_output)
>       self.assert_dirs_equal('input', 'output/input')

src/borg/testsuite/archiver.py:420: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/borg/testsuite/__init__.py:172: in assert_dirs_equal
    self._assert_dirs_equal_cmp(diff, **kwargs)
src/borg/testsuite/__init__.py:218: in _assert_dirs_equal_cmp
    self.assert_equal(d1, d2)
E   AssertionError: Lists differ: ['link1', '0o120755', 1000, 0, None, 1, 0, 1557601102849398658, {}] != ['link1', '0o120700', 1000, 0, None, 1, 0, 1557601102849398658, {}]
E   
E   First differing element 1:
E   '0o120755'
E   '0o120700'
E   
E   - ['link1', '0o120755', 1000, 0, None, 1, 0, 1557601102849398658, {}]
E   ?                  ^^
E   
E   + ['link1', '0o120700', 1000, 0, None, 1, 0, 1557601102849398658, {}]
E   ?                  ^^
========================================================================= 1 failed, 3 passed, 1 skipped, 1348 deselected in 21.68 seconds 
```

--- a/src/borg/archive.py
+++ b/src/borg/archive.py
@@ -49,7 +49,6 @@ from .platform import acl_get, acl_set, set_flags, get_flags, swidth, hostname
 from .remote import cache_if_remote
 from .repository import Repository, LIST_SCAN_LIMIT
 
-has_lchmod = hasattr(os, 'lchmod')
 has_link = hasattr(os, 'link')
 
 
@@ -877,10 +876,18 @@ Utilization of max. archive size: {csize_max:.0%}
                 pass
             if fd:
                 os.fchmod(fd, item.mode)
-            elif not symlink:
-                os.chmod(path, item.mode)
-            elif has_lchmod:  # Not available on Linux
-                os.lchmod(path, item.mode)
+            else:
+                # To check whether a particular function in the os module accepts False for its
+                # follow_symlinks parameter, the in operator on supports_follow_symlinks should be
+                # used. However, os.chmod is special as some platforms without a working lchmod() do
+                # have fchmodat(), which has a flag that makes it behave like lchmod(). fchmodat()
+                # is ignored when deciding whether or not os.chmod should be set in
+                # os.supports_follow_symlinks. Work around this by using try/except.
+                try:
+                    os.chmod(path, item.mode, follow_symlinks=False)
+                except NotImplementedError:
+                    if not symlink:
+                        os.chmod(path, item.mode)
             mtime = item.mtime
             if 'atime' in item:
                 atime = item.atime

